doc/sphinx/src/particles.rst

```diff
@@ -161,3 +161,17 @@ Swarm::bounds array entries to factory functions that allocate
 device-side boundary condition objects. An example is given in the
 ``particles`` example when ix1 and ox1 are set to ``user`` in the input
 parameter file.
+
+Outputs
+--------
+
+Outputs for swarms can be set in an output block, just like any other
+variable. The user must specify two comma separated lists:
+
+* ``swarms = swarm1, swarm2, ...`` etc. This specifies that these swarms
+  are marked for output.
+
+* ``swarm_variables = var1, var2, ...`` etc. This specifies that for
+  each swarm the listed variables should be output. Note that if a
+  swarm is missing a variable in the ``swarm_variables`` list, no
+  error is raised. All available variables are output.
```

Comment:
This should always produce the requested data in output files but for multiple swarms this could easily produce too much data, I think. 

One alternative would be to allow e.g. `<parthenon/output0/swarm0>` blocks that internally contained the swarm name and the vars to output e.g.
```
<parthenon/output0/swarm0>
swarm = my particles
variables = id, v, vv
```

This is something I could do subsequently, except that since it would be a breaking change maybe we should nail down the "right" way to do things here?

Reply:
Hmm... that's kind of hard because it requires walking a second output block, which the current `Output` class isn't really designed to do. What about this?
```
<parthenon/output0>
swarm = my_particles #swarms you want to output
# per swarm a list of variables
my_particles = id, v, vv
```

Alternatively, what about this?
```
<parthenon/output0>
swarm = my_particles #swarms you want to output

<parthenon/output0/my particles>
variables= id, v, vv
```
would one of those work? And if so which do you think you like better?

Reply:
The former one requires forbidding spaces in swarm names, but maybe that's a good thing anyway.

Reply:
Ah ok -- I have a slight preference for the latter since I think it is a bit more explicit. If the former is substantially easier I think that is fine though.

One could modify the first one probably to be
```
<parthenon/output0>
swarm = my_particles #swarms you want to output
# per swarm a list of variables
my_particles_variables = id, v, vv
```
without much extra effort I would think? That's a bit more explicit.

Regardless, prohibiting spaces in swarm names is probably a good thing.

Reply:
I think `<parthenon/output0/my particles>` and
 ```
<parthenon/output0>
my_particles_variables
```
are roughly the same difficulty. I just need to list the swarm names somewhere in `parthenon/output0`, not elsewhere.

Reply:
Both approaches feel a bit weird. But maybe I will go with `<parthenon/output0/my particles>` as that feels more consistent with parthenon patterns.

Reply:
Ok turns out `<parthenon/output0/my particles>` solution doesn't work without some surgery because of the way the code walks through output blocks.  So I went with the `my_particles_variables` solution.

I did not, however, remove the `swarm_variables` input field. The way it works now is that all variables listed in `swarm_variables` are assumed to be requested by all swarms, whereas variables in the  `my_swarm_name_variables` field are only for `my_swarm_name`. I think this is a good compromise that should offer a nice level fidelity.

Reply:
OK yeah I agree that's a good compromise, makes sense and exposes maximum granularity in requesting data output -- thanks

Reply:
this did require forbidding spaces in swarm names and also forbidding any swarm from just being named "swarm," since that would cause a name collision

Reply:
I think both those restrictions are good to have in place